syntax = "proto3";

package code_editor;

service CodeAgent {
  // Generate code changes and Submit pull request to fix given issue
  rpc FixIssue (IssueFixRequest) returns (IssueFixResponse) {}
}

message IssueFixRequest {
  string full_repo_name = 1;

  Issue issue = 2;
}

message Issue {
  string title = 1;
  string body = 2;
}

message IssueFixResponse {
  string pull_request_url = 1;
  //TODO add status code
}